### ISSUE TITLE
Problem with integer number parsing

### ISSUE THREAD

**Issue by ghost** [r1chardj0n3s/parse#28]
Hi there. 
I'm having a problem with the 1.6.6 version in both python 2.7.9 and python 3.4.3. 

I was able to reproduce it with the following example:

``` python
>>> import parse
>>> parse.parse("blablabla {x:d}", "blablabla 12")
<Result () {'x': 12}>
>>> parse.parse("blablabla {x:d}", "blablabla jdhhd")
>>> parse.parse("blablabla {x:d}", "blablabla cdc")                                                                                                                                                                                
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/calsaverini/.envs/default/lib/python3.4/site-packages/parse.py", line 1044, in parse
    return Parser(format, extra_types=extra_types).parse(string)
  File "/home/calsaverini/.envs/default/lib/python3.4/site-packages/parse.py", line 681, in parse
    return self._generate_result(m)
  File "/home/calsaverini/.envs/default/lib/python3.4/site-packages/parse.py", line 731, in _generate_result
    m)
  File "/home/calsaverini/.envs/default/lib/python3.4/site-packages/parse.py", line 398, in f
    return sign * int(string, base)
ValueError: invalid literal for int() with base 10: ''
```

Is this a bug or am I missing some kind of edge case when `{:d}` might match `cdc` that I'm not aware of? 

Thanks for your help.

**Comment by r1chardj0n3s** [comment]
At first glance, "cdc" are valid letters in a hexadecimal number. Hm.

**Comment by Ricyteach** [comment]
I have also noticed this unexpected behavior: 

    >>> parse('{}{:d}', 'abcd1')
    <Result ('a', 1) {}>

However: 

    >>> parse('{}{:d}', 'abcdefg1')
    <Result ('abcdefg', 1) {}>

I would say the hexadecimal theory is likely correct since the problem goes away for any letter after "f".

**Comment by r1chardj0n3s** [comment]
I believe this issue was resolved in commit 2918b126daffc9bb20ec7a6f569d3a21e0378b2d which was released as version 1.8.1

**Comment by Ricyteach** [comment]
Turns out you are correct!
